lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,6 +36,15 @@ def to_native(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            course_overview_tabs = course_overview.course_overview_tab.all()
+            for tab in course_overview_tabs:
+                if tab.tab_id == "discussion" and is_discussion_enabled(course_id):
```

Comment:
Consider moving this logic as a helper method in the `CourseOverviewTab` code.  It seems generic enough that it could be used by other callers in the future.  (For example, the discussion_api would probably want to use `CourseOverview` rather than `CourseDescriptor` soon for performance reasons.)